packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -85,123 +109,506 @@ const propTypes = {
+      inputRefCallback(node);
+    }
+  }, [dateFormatOrder, inputRefCallback]);
+
+  const setMonthRef = useCallback(node => {
+    monthInputRef.current = node;
+  }, []);
+
+  const setYearRef = useCallback(node => {
+    yearInputRef.current = node;
+    if (inputRefCallback && (dateFormatOrder === DateUtil.dateOrder.MDY || dateFormatOrder === DateUtil.dateOrder.DMY)) {
+      inputRefCallback(node);
+    }
+  }, [dateFormatOrder, inputRefCallback]);
+
+  useEffect(() => {
+    if (shouldShowPicker && onClick) {
+      onClick();
+    }
+  }, [shouldShowPicker, onClick]);
+
+  // Sets the date state based on the passed in value prop, or if it changes via a calendar click.
+  useEffect(() => {
+    if (DateUtil.isValidDate(value, momentDateFormat)) {
+      const parseDate = DateUtil.getDateInputValues(DateUtil.dateOrder.YMD, dateValue, '-');
+      dateDispatch(parseDate);
+    } else if (dateValue === '') {
+      dateDispatch({ day: '', month: '', year: '' });
+    }
+  }, [momentDateFormat, dateValue, value]);
+
+  /**
+   * Moves focus to the correct input depending on date ordering. Focus changing is
+   * disabled if a complete date has been entered in order to make single input
+   * corrections easier, and is re-enabled if the whole date is erased.
+   * @param {string} inputValue - The value from the current input.
+   * @param {number} type - The input type, based on DateUtil.inputType.
+   * @param {string} dateOrder - Date variant based on the currently supported date orders: 'MDY', 'DMY', or 'YMD'.
+   */
+  const moveFocusOnChange = (inputValue, type, dateOrder) => {
+    if (dateOrder === DateUtil.dateOrder.MDY) {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.MONTH) {
+          dayInputRef.current.focus();
+        } else {
+          yearInputRef.current.focus();
+        }
+      }
+    } else if (dateOrder === DateUtil.dateOrder.DMY) {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.DAY) {
+          monthInputRef.current.focus();
+        } else {
+          yearInputRef.current.focus();
+        }
+      }
+    } else if (dateOrder === DateUtil.dateOrder.YMD) {
+      if (inputValue.length === 4) {
+        monthInputRef.current.focus();
+      } else if (inputValue.length === 2 && type === DateUtil.inputType.MONTH) {
+        dayInputRef.current.focus();
+      }
+    }
+  };
+
+  /**
+   * Sets state for the day, month, and year.
+   * @param {string} inputValue - The input value to set in state.
+   * @param {number} type - The inputType (day, month, or year).
+   */
+  const setDate = (inputValue, type) => {
+    if (type === DateUtil.inputType.DAY) {
+      dateDispatch({ day: inputValue });
+    } else if (type === DateUtil.inputType.MONTH) {
+      dateDispatch({ month: inputValue });
+    } else {
+      dateDispatch({ year: inputValue });
+    }
+
+    moveFocusOnChange(inputValue, type, dateFormatOrder);
```

Comment:
Can `moveFocusOnChange` just get the `dateFormatOrder` directly instead of having to pass it since it is just a constant declared above?

Reply:
Updated here: https://github.com/cerner/terra-framework/pull/1208/commits/623265a07fb2323e62a269e3e72b183ee8c8b2da